ISISScientificComputing/autoreduce-rest-api
# Implements initial version of REST API

### Summary of work
Implements initial version with:
- POST request for submission
- DELETE request for deleting runs
- Token authentication (with tokens generated from https://github.com/ISISScientificComputing/autoreduce-frontend/pull/27)
 
### How to test your work
Tests should pass.

[Start up](https://github.com/ISISScientificComputing/autoreduce/wiki/Start-here-(getting-the-code---installation)) the whole service (with ActiveMQ & queue processor), and submit runs to the API. You can use Postman or just curl requests like so:

```
curl --location --request POST 'http://127.0.0.1:8001/api/runs/INTER/63125' \
--header 'Authorization: Token <token>'

curl --location --request DELETE 'http://127.0.0.1:8001/api/runs/INTER/63125' \
--header 'Authorization: Token <token>'
```

### Additional comments
